Userland/Demos/CatDog/main.cpp

```diff
@@ -54,6 +68,39 @@ int main(int argc, char** argv)
     catdog_widget.layout()->set_spacing(0);
 
     auto context_menu = GUI::Menu::construct();
+    g_skins_group.set_exclusive(true);
+    g_skins_group.set_unchecking_allowed(false);
+
+    g_skins_menu = &context_menu->add_submenu("Skins");
+
+    {
+        Core::DirIterator dt("/res/icons/catdog", Core::DirIterator::SkipDots);
+        while (dt.has_next()) {
+            auto skin_name = dt.next_path();
+            auto skin_path = String::formatted("/res/icons/catdog/{}", skin_name);
+            g_skins.append({ LexicalPath::title(skin_name), skin_path });
+            dbgln("hi {} at {}", skin_name, skin_path);
+        }
+        quick_sort(g_skins, [](auto& a, auto& b) { return a.name < b.name; });
+    }
+
+    auto current_skin_name = catdog_widget.get_skin_name();
+
+    {
+        int skin_identifier = 0;
+        for (auto& skin : g_skins) {
+            auto action = GUI::Action::create_checkable(skin.name, [&, skin_identifier](auto&) {
+                auto& skin = g_skins[skin_identifier];
+                dbgln("Skin switched to {} at path {}", skin.name, skin.path);
+                catdog_widget.set_skin(skin.name, skin.path);
```

Comment:
Perhaps this should take a `SkinMetadata` instead? Seems odd to group the values in a struct here but not elsewhere.

Reply:
I based this part of the code from the startmenu's themes picker, so it is more less the same as that one.